### ISSUE TITLE
add a Java 9 community build

### ISSUE THREAD

**Issue by SethTisue** [scala/community-builds#609]
starting, I suppose, with just a stripped-down thing that only attempts to build some basic stuff like the Scala standard modules, then expanding from there

Scala 2.12 only is fine for now, I think. the behemoths are already somewhat pressed for space (and the space can't be increased without rebuilding the nodes), and even just adding the 2.12/jdk9 combo will make the problem, one might say, worse enough

plan:
* get it running locally on my Mac
* get it running on Jenkins

re: Jenkins, @retronym did some work, prior to JDK 9 going final/gold/GA, on making it available on our Jenkins nodes; see https://github.com/scala/scala-jenkins-infra/issues/174 for details

**Comment by SethTisue** [comment]
my experience attempting this locally was that all extractions failed with `[scalacheck] [error] scala.reflect.internal.MissingRequirementError: object java.lang.Object in compiler mirror not found.`

but then I made an empty sbt project and ran it with Java 9, and https://github.com/sbt/sbt-launcher-package/pull/143 kicked in and showed the "Copying runtime jar." message. I thought then running dbuild again might work, but same error

maybe I also need to do something to make sure `-Dscala.ext.dirs=...` gets set?

**Comment by SethTisue** [comment]
> maybe I also need to do something to make sure -Dscala.ext.dirs=... gets set?

helps, but is not sufficient. adding `extra.options: ["-Dscala.ext.dirs=/Users/tisue/.sbt/0.13/java9-rt-ext-oracle_corporation_9"]` to every projects allows extraction to proceed farther, but then eventually every projects gets an error like:

```text
[scala-parser-combinators] java.lang.AssertionError: assertion failed: not a type: symbol
[scala-parser-combinators]  loaded from class ScalaModulePlugin$ in package with name ScalaModulePlugin$ and classloader sbt.PluginManagement$PluginClassLoader@9b8d3db
[scala-parser-combinators] 	at scala.reflect.runtime.JavaMirrors$JavaMirror.scala$reflect$runtime$JavaMirrors$JavaMirror$$classToScala1(JavaMirrors.scala:979)
[scala-parser-combinators] 	at scala.reflect.runtime.JavaMirrors$JavaMirror$$anonfun$classToScala$1.apply(JavaMirrors.scala:938)
[scala-parser-combinators] 	at scala.reflect.runtime.JavaMirrors$JavaMirror$$anonfun$classToScala$1.apply(JavaMirrors.scala:938)
```

**Comment by SethTisue** [comment]
an old (2014) ticket on Java 9 support dbuild is https://github.com/typesafehub/dbuild/issues/155

**Comment by retronym** [comment]
That might be https://github.com/scala/scala-dev/issues/304. The only workaround I know it to refactor the `.scala` files in the build to avoid using the empty package.

**Comment by SethTisue** [comment]
> java.lang.AssertionError: assertion failed: not a type: symbol

this is reproducible outside dbuild simply by switching to Java 9 and then firing up sbt in e.g. the scala-xml or scala-parser-combinators repo

two repos that *don't* have this problem: utest, scalacheck

as an experiment, I went into scala-parser-combinators and ripped out sbt-scala-module, after which `scala-parser-combinatorsJVM/test` passed. so the problem is with that plugin somehow

**Comment by SethTisue** [comment]
> the problem is with that plugin somehow

okay, so let's work on that over at https://github.com/scala/sbt-scala-module/issues/33, then come back here later

**Comment by SethTisue** [comment]
a "community build" consisting only of scalacheck (with scala and scala-xml binaries retrieved via aether), at https://github.com/SethTisue/community-builds/blob/wip-2.12.x-jdk9/configs/community.dbuild, is passing locally with the `extra.options: ["-Dscala.ext.dirs=...` workaround 🎉

so it looks like dbuild works on Java 9, which is good news, and we'll just need to expand the build a step at a time from here (and work around the `scala.ext.dirs` thing in a more elegant way)

**Comment by SethTisue** [comment]
> we'll just need to expand the build a step at a time from here

questioning, all the while, how ambitious it's appropriate to get with this how soon! I could see this staying pretty minimal for a long time

until _at least_:

* community projects have added Java 9 to their own CI matrices
    * which will probably be happening pretty rapidly over the next year or so... but in the meantime we don't need to get out ahead of the community on this

and probably not even until either or both of these stronger conditions are met:

* we start work on a version of Scala which requires Java 9
* significant community projects start requiring Java 9
    * this is what prompted the creation of the `scala-2.11.x-jdk8` build — significant libraries started dropping support for Java 6 and/or 7

**Comment by retronym** [comment]
Are we able to set the Java version individually per-project? That would allow us more flexibility: we could build problematic projects on Java 8 and run compliant ones on 9.

**Comment by cunei** [comment]
Not at the moment; dbuild uses whatever version of Java is in the environment, and makes no attempt to control it.

**Comment by SethTisue** [comment]
Akka has announced Java 9 support (as of Akka 2.5.6). I'll make Akka a priority next time I do some work on this

**Comment by SethTisue** [comment]
the new build has grown from one to five projects.

they are: scala-xml, plus

```
[info] Project scala-js----------------: SUCCESS (project rebuilt ok)
[info] Project scala-parser-combinators: SUCCESS (unchanged, not rebuilt)
[info] Project scalacheck--------------: SUCCESS (unchanged, not rebuilt)
[info] Project utest-------------------: SUCCESS (project rebuilt ok)
```

how I got there:

I rebased Jason's backport PR onto 2.10.x (which has other JDK9 stuff merged on it that Jason's PR doesn't include yet)

after much trial and error, my efforts to get dbuild+sbt to use a Scala version with "SNAPSHOT" and/or a SHA in the version number failed, so I publish a 2.10.7 snapshot as `2.10.7` to `~/.m2` locally (no `-SNAPSHOT` or SHA in the version):

```
brew install ant@1.9
PATH=/usr/local/Cellar/ant@1.9/1.9.9/bin:$PATH
export JAVA_HOME=$(/usr/libexec/java_home -v 1.6)
ANT_OPTS="-Xmx2048M -Xss4M -XX:MaxPermSize=256M" ant -Dbuild.release=true fastdist distpack
cd dists/maven/2.10.7
ant deploy.local
```

and then in my repositories file put:

```
  maven-local-alt: file:///Users/tisue/.m2/repository/
```

since as @dwijnand discovered, sbt is hardwired to ignore a normal `maven-local` entry when resolving Scala snapshots specifically (?!)

the config file changes necessary are in the wip branch in my fork

I'm also not going to bother trying to run this on Jenkins until Scala 2.10.7 and sbt 0.13.17 are published — I think it would be wasted work, it's enough if I just run it locally for now.

**Comment by SethTisue** [comment]
@retronym might pick this up and run with it for a while. thus, brain dump:

I have pushed my wip fumblings on this to https://github.com/SethTisue/community-builds/commits/wip-2.12.x-jdk9

to get it up on Jenkins, some work on the scala-jenkins-infra repository will be needed:
* the new build should go on behemoth 1; behemoth 2 already runs out of space a lot, behemoth 1 is much less prone to that. (we have so far avoided just cranking up the disk space on the boxes because doing so requires re-creating them, which Chef is only some help with)
* an appropriate JDK needs to be available. Jason already did some work to make JDK 9 available there, I think, but it might be out of date? ideally we go through Chef to get stuff installed on our Jenkins nodes, but if doing so is just too much hassle, we're not above just manually installing something; I think we have done that there in the past with JDKs. @adriaanm might be more aware than I am of where we currently stand on that.

adding more Scala modules to the build would require doing something about https://github.com/scala/sbt-scala-module/issues/35, iirc

besides the remaining Scala modules, obvious next additions could include ScalaTest and akka-actor

**Comment by SethTisue** [comment]
sbt 0.13.17-RC2 and 1.1.0 exist now and should make this a bit easier

**Comment by SethTisue** [comment]
what are those great crashing footfalls I hear? it's our THREE mighty behemoths, LARGER than the old ones. what a thrilling and awe-inspiring sight they are! and apparently they have JDK 9 on them already 🤖🤖🤖 💪💪💪 ⓽⓽⓽ 💥💥💥

I created https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-jdk9-integrate-community-build via the Jenkins UI by copying the regular 2.12.x job, adding `-jdk9-` to the name, and changing the JVM version to 9, and a run is in progress here: https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-jdk9-integrate-community-build/1/consoleFull

note that's a run of the regular 2.12.x branch of this repo. thought I'd see how that does first. we can also try the `wip-2.12.x-jdk9` branch — some of the changes there shouldn't be needed anymore, maybe all of them.

**Comment by ashawley** [comment]
It failed.  Didn't look like there were any more builds kicked off.  Did this end up getting hung up?

**Comment by SethTisue** [comment]
work on 2.12.5 (and the usual thousand other things) interceded. ironically, since this would have caught the big regression in 2.12.5. sigh

yeah so in the first run a bunch of projects failed to even extract, I looked at a sampling and it was usually (always?) because something in the build choked on the unexpected Java version number

so I'll need to start just marching down the list and making pull requests or issues in those projects, starting with the ones with the most downstream dependents

I kicked off a new run just now at https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-jdk9-integrate-community-build/2/console , perhaps some of the included projects have already been fixed

**Comment by ashawley** [comment]
> it was usually (always?) because something in the build choked on the unexpected Java version number

Oh, ok. Yeah, that's brutal.

**Comment by retronym** [comment]
```
[argonaut-shapeless] [error] (coursierResolutions) java.lang.IncompatibleClassChangeError: Inconsistent constant pool data in classfile for class coursier/shaded/fastparse/core/LowestPriSequencer. Method $anonfun$Sequencer1$1(Ljava/lang/Object;Ljava/lang/Object;)Lscala/Tuple2; at index 39 is CONSTANT_MethodRef and should be CONSTANT_InterfaceMethodRef
[argonaut-shapeless] [error] (coursierSbtClassifiersResolution) java.lang.IncompatibleClassChangeError: Inconsistent constant pool data in classfile for class coursier/shaded/fastparse/core/LowestPriSequencer. Method $anonfun$Sequencer1$1(Ljava/lang/Object;Ljava/lang/Object;)Lscala/Tuple2; at index 39 is CONSTANT_MethodRef and should be CONSTANT_InterfaceMethodRef
[argonaut-shapeless] Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? 
```

I know this problem, it requires an updated shade plugin, as in https://github.com/playframework/play-ws/issues/173. I'll take this up with coursier and argonaut-shapeless.

Issue: https://github.com/coursier/coursier/issues/822

Workaround: https://github.com/scala/community-builds/pull/700

**Comment by retronym** [comment]
```
[paradox] [info] 'compiler-interface' not yet compiled for Scala 2.10.6. Compiling...
[paradox:error] error: error while loading package, Missing dependency 'object java.lang.Object in compiler mirror', required by /home/jenkins/.ivy2/cache/org.scala-lang/scala-library/jars/scala-library-2.10.6.jar(scala/package.class)
[paradox:error] error: error while loading package, Missing dependency 'object java.lang.Object in compiler mirror', required by /home/jenkins/.ivy2/cache/org.scala-lang/scala-library/jars/scala-library-2.10.6.jar(scala/runtime/package.class)
[paradox:error] error: scala.reflect.internal.MissingRequirementError: object java.lang.Object in compiler mirror not found.
[paradox:error] 	at scala.reflect.internal.MissingRequirementError$.signal(MissingRequirementError.scala:16)
[paradox:error] 	at scala.reflect.internal.MissingRequirementError$.notFound(MissingRequirementError.scala:17)
[paradox:error] 	at scala.reflect.internal.Mirrors$RootsBase.getModuleOrClass(Mirrors.scala:48)
[paradox:error] 	at scala.reflect.internal.Mirrors$RootsBase.getModuleOrClass(Mirrors.scala:40)
```

This probably just means that project is building with SBT 0.13.16 or lower, which is yet to update to Scala 2.10.7 to get support of for Java 9.

https://github.com/scala/community-builds/pull/701

**Comment by SethTisue** [comment]
> This probably just means that project is building with SBT 0.13.16 or lower

that's odd — we override the sbt versions of all included projects (to 0.13.17 and 1.1.1, currently)

**Comment by ashawley** [comment]
I've opened a PR for ScalaTest:

https://github.com/scalatest/scalatest/pull/1340

**Comment by SethTisue** [comment]
this is still a back-burner task, but, to at least make sure we're not slipping backwards, I merged current 2.12.x onto https://github.com/SethTisue/community-builds/commits/wip-2.12.x-jdk9 and made sure it still works:

```
[info] Project cloc-plugin-----------------: SUCCESS (unchanged, not rebuilt)
[info] Project scala-----------------------: SUCCESS (unchanged, not rebuilt)
[info] Project scala-js--------------------: SUCCESS (project rebuilt ok)
[info] Project scala-js-stubs--------------: SUCCESS (project rebuilt ok)
[info] Project scala-parser-combinators----: SUCCESS (unchanged, not rebuilt)
[info] Project scalacheck------------------: SUCCESS (unchanged, not rebuilt)
[info] Project utest-----------------------: SUCCESS (project rebuilt ok)
[info] >>> The dbuild result is----------------: SUCCESS (project rebuilt ok)
```

as for the full build, I've set a personal to-do item to make gradual progress on at least getting all of the projects extracting. today's increment of progress on that is here: https://github.com/scala/scala-swing/pull/78

**Comment by SethTisue** [comment]
another increment: https://github.com/milessabin/shapeless/pull/830

**Comment by SethTisue** [comment]
today's increment of progress: akka-http needs the same sbt-api-mappings version bump I submitted to Shapeless. it has the version bump already on master, we're frozen at an older commit. not sure how easy unfreezing will be, but let's try it: https://github.com/scala/community-builds/pull/723

UPDATE: yup, akka-http now extracting at https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-jdk9-integrate-community-build/36/

**Comment by SethTisue** [comment]
today's increment of progress: https://github.com/scalanlp/breeze/pull/708

**Comment by ashawley** [comment]
I opened a PR for ScalaCheck:

https://github.com/rickynils/scalacheck/pull/407

**Comment by SethTisue** [comment]
scalameter and paradox are both failing to extract because they have meta-builds that aren't JDK 9 friendly

scalameter's meta-build in `project/project/Build.scala` pulls in https://github.com/storm-enroute/mecha which sets the Scala version to 2.10.4

and paradox's meta-build in `project/plugins.sbt` has `scalaVersion := "2.10.6"`

figuring that much out is my increment of progress for today. I'll work on fixes/workarounds next.

**Comment by SethTisue** [comment]
today's increment of progress: opened these PRs:

* https://github.com/lightbend/paradox/pull/197
* https://github.com/storm-enroute/mecha/pull/8

I'm also tempted to learn how to use dbuild to tweak things in the meta-build... but if these PRs are merged, then that works too

**Comment by SethTisue** [comment]
today's increment of progress: fixed up and merged https://github.com/lightbend/paradox/pull/197, and then merged the changes onto the branch we're using in our fork

[current](https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-jdk9-integrate-community-build/51/console) status: breeze (awaiting PR merge), monix (?), pureconfig (?), scala-debugger (?), scalameter (mecha PR was merged but that doesn't seem to have fixed it), shapeless (awaiting PR merge)

**Comment by SethTisue** [comment]
scalameter extracted in a local run, so I thought `git clean -ffdx` in `~/workspace/scala-2.12.x-jdk9-integrate-community-build` on the worker might fix it on Jenkins too, and [it did!](https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-jdk9-integrate-community-build/52/consoleFull)

**Comment by SethTisue** [comment]
today's progress: dropped scala-debugger from the build in #725 (but @-mentioned its author and asked if he's interested in helping it get back in)

**Comment by SethTisue** [comment]
scala-debugger is now reinstated, Chip made the needed fix

**Comment by SethTisue** [comment]
monix PR: https://github.com/monix/monix/pull/671. but the PR being merged won't fix it on our end because we're frozen at an old SHA, so I've also forked it

pureconfig: we already have it forked, so I just pushed SethTisue/pureconfig@36d299fe43c022443445c00e4f36e6f384bb1d91

**Comment by SethTisue** [comment]
current status: still waiting for these PRs to be merged: https://github.com/scalanlp/breeze/pull/708 (ping @dlwh!), https://github.com/milessabin/shapeless/pull/830 (ping @milessabin!), otherwise we're good to go

**Comment by SethTisue** [comment]
scalameter stopped extracting but clearing the workspace again fixed it, I think because (again) the source dependency on mecha needed refreshing.

as for breeze and shapeless, 11fb142480bf96d2cd998d4a11950fa43342d9df uses the PR branches, I'll revert once the PRs are merged.

https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-jdk9-integrate-community-build/112/consoleFull

**Comment by SethTisue** [comment]
yay everything extracted 🎉

now some OSGI thing in scala-xml is barfing, and then of course nothing else can run

`[scala-xml] [error] (xmlJVM/*:osgiRequireCapability) java.lang.StringIndexOutOfBoundsException: begin 6, end -1, length 14`

<details>

```
[scala-xml] [info] Done packaging.
[scala-xml] java.lang.StringIndexOutOfBoundsException: begin 6, end -1, length 14
[scala-xml] 	at java.base/java.lang.String.checkBoundsBeginEnd(String.java:3116)
[scala-xml] 	at java.base/java.lang.String.substring(String.java:1885)
[scala-xml] 	at com.typesafe.sbt.osgi.Osgi$$anon$1.log(Osgi.scala:69)
[scala-xml] 	at sbt.LoggerWriter$$anonfun$flushLines$1.apply(LoggerWriter.scala:28)
[scala-xml] 	at sbt.LoggerWriter$$anonfun$flushLines$1.apply(LoggerWriter.scala:27)
[scala-xml] 	at scala.collection.immutable.List.foreach(List.scala:318)
[scala-xml] 	at scala.collection.generic.TraversableForwarder$class.foreach(TraversableForwarder.scala:32)
[scala-xml] 	at scala.collection.mutable.ListBuffer.foreach(ListBuffer.scala:45)
[scala-xml] 	at sbt.LoggerWriter.flushLines(LoggerWriter.scala:27)
[scala-xml] 	at sbt.compiler.JavaCompiler$$anonfun$2.apply(JavaCompiler.scala:134)
[scala-xml] 	at sbt.compiler.JavaCompiler$$anonfun$2.apply(JavaCompiler.scala:125)
[scala-xml] 	at sbt.compiler.JavaCompiler$$anonfun$directOrForkJavac$1.apply(JavaCompiler.scala:110)
[scala-xml] 	at sbt.compiler.JavaCompiler$$anonfun$directOrForkJavac$1.apply(JavaCompiler.scala:109)
[scala-xml] 	at sbt.compiler.JavaCompiler$JavaTool0.compile(JavaCompiler.scala:95)
[scala-xml] 	at sbt.Compiler$CheaterJavaTool$6.compile(Compiler.scala:84)
[scala-xml] 	at sbt.compiler.JavaTool$class.apply(JavaCompiler.scala:54)
[scala-xml] 	at sbt.Compiler$CheaterJavaTool$6.apply(Compiler.scala:82)
[scala-xml] 	at com.typesafe.sbt.osgi.Osgi$.version$1(Osgi.scala:66)
[scala-xml] 	at com.typesafe.sbt.osgi.Osgi$.requireCapabilityTask(Osgi.scala:76)
[scala-xml] 	at com.typesafe.sbt.osgi.SbtOsgi$$anonfun$defaultOsgiSettings$12.apply(SbtOsgi.scala:83)
[scala-xml] 	at com.typesafe.sbt.osgi.SbtOsgi$$anonfun$defaultOsgiSettings$12.apply(SbtOsgi.scala:83)
[scala-xml] 	at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
[scala-xml] 	at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
[scala-xml] 	at sbt.std.Transform$$anon$4.work(System.scala:63)
[scala-xml] 	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
[scala-xml] 	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
[scala-xml] 	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
[scala-xml] 	at sbt.Execute.work(Execute.scala:237)
[scala-xml] 	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
[scala-xml] 	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
[scala-xml] 	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
[scala-xml] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
[scala-xml] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[scala-xml] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:514)
[scala-xml] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[scala-xml] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
[scala-xml] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
[scala-xml] 	at java.base/java.lang.Thread.run(Thread.java:844)
[scala-xml] [error] (xmlJVM/*:osgiRequireCapability) java.lang.StringIndexOutOfBoundsException: begin 6, end -1, length 14
[scala:error] java.lang.RuntimeException: 
[scala:error] 	at scala.sys.package$.error(package.scala:27)
[scala:error] 	at com.typesafe.dbuild.support.sbt.SbtRunner.com$typesafe$dbuild$support$sbt$SbtRunner$$processCommand(SbtRunner.scala:94)
[scala:error] 	at com.typesafe.dbuild.support.sbt.SbtRunner$$anonfun$run$1$$anonfun$apply$2$$anonfun$apply$3.apply(SbtRunner.scala:75)
[scala:error] 	at com.typesafe.dbuild.support.sbt.SbtRunner$$anonfun$run$1$$anonfun$apply$2$$anonfun$apply$3.apply(SbtRunner.scala:75)
[scala:error] 	at com.typesafe.dbuild.support.sbt.SbtRunner$$anonfun$run$1.apply(SbtRunner.scala:75)
[scala:error] 	at com.typesafe.dbuild.support.sbt.SbtRunner$$anonfun$run$1.apply(SbtRunner.scala:66)
[scala:error] 	at sbt.IO$.withTemporaryFile(IO.scala:389)
[scala:error] 	at com.typesafe.dbuild.support.sbt.SbtRunner.run(SbtRunner.scala:66)
[scala:error] 	at com.typesafe.dbuild.support.sbt.SbtBuilder$.buildSbtProject(SbtBuildRunner.scala:93)
[scala:error] 	at com.typesafe.dbuild.support.sbt.SbtBuildSystem.runBuild(SbtBuildSystem.scala:96)
[scala:error] 	at com.typesafe.dbuild.support.sbt.SbtBuildSystem.runBuild(SbtBuildSystem.scala:22)
```

</details>

**Comment by SethTisue** [comment]
maybe https://github.com/scala/scala-xml/pull/218 will help

**Comment by SethTisue** [comment]
it didn't, and I couldn't reproduce it outside dbuild, so maybe it's choking on a dbuild-mangled version number

**Comment by SethTisue** [comment]
gah, of course, it's https://github.com/scala/sbt-scala-module/issues/35 again, which has been pending so long I forgot about it

**Comment by SethTisue** [comment]
https://github.com/scala/scala-xml/pull/220 should do it

**Comment by SethTisue** [comment]
that worked. stuff is building now. let's see how far it gets: https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-jdk9-integrate-community-build/115/consoleFull

**Comment by SethTisue** [comment]
sigh, for some reason https://github.com/slick/slick/issues/1908 brings the entire build to a halt

**Comment by SethTisue** [comment]
do the best you can, [run 117](https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-jdk9-integrate-community-build/117/)!

**Comment by hvesalai** [comment]
You need to add `--add-modules java.xml.bind -XX:+IgnoreUnrecognizedVMOptions` to JVM parameters

**Comment by SethTisue** [comment]
thanks, that does it (`--add-modules=java.xml.bind`). next up: [run 118](https://scala-ci.typesafe.com/job/scala-2.12.x-jdk9-integrate-community-build/118/)

**Comment by SethTisue** [comment]
latest is [run 121](https://scala-ci.typesafe.com/job/scala-2.12.x-jdk9-integrate-community-build/121/)

in my local testing, we're now very close to being able to declare victory (as in, enough green projects, not all of them), but there is one more keystone project that is failing and taking out a big swath of the build, namely specs2. I've filed https://github.com/etorreborre/specs2/issues/659 on it

**Comment by SethTisue** [comment]
oh and there's one other thing I need to fix, which is that I pushed a "fix" for Scala.js that hardcodes `/Users/tisue` which obviously doesn't work on Jenkins. I think we shouldn't need to be setting `scala.ext.dirs` explicitly like this, it should be set for us by the sbt launch script, except I guess on Jenkins we have an outdated one.

locally, with Homebrew-installed sbt, if I  run sbt on Java 9 and do `eval System.getProperty("scala.ext.dirs")` I get a path, but if I grab `/usr/local/bin/sbt` off of Jenkins (it's the sbt-extras one, not sure what version) and run that locally, `scala.ext.dirs` is `null`

**Comment by SethTisue** [comment]
specs2 needs scala-js-stubs, which I haven't fixed yet (still the highest priority needed fix)

scala-collection-compat and sourcecode were both failing because of a too-old sbt-osgi version; I've pushed fixes for both

**Comment by SethTisue** [comment]
I pushed a scala-js-stubs fix. https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-jdk9-integrate-community-build/182/consoleFull has scala-js-stubs green. the rest is still running, I'll swing back by tomorrow and see how it did.

**Comment by SethTisue** [comment]
119 projects are green now! declaring victory.

**Comment by SethTisue** [comment]
* https://github.com/non/jawn/pull/106 (PR)
* https://github.com/pathikrit/better-files/issues/242 (issue requesting help)

**Comment by SethTisue** [comment]
* https://github.com/unfiltered/unfiltered/issues/418 (issue reporting test failures)

**Comment by SethTisue** [comment]
* https://github.com/scalapb/ScalaPB/issues/466 (issue requesting fix)

**Comment by SethTisue** [comment]
* https://github.com/scodec/scodec-bits/pull/92 (PR with fix)

**Comment by SethTisue** [comment]
status at https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-jdk9-integrate-community-build/274/ : `[info] >>> The dbuild result is----------------: FAILED (failed: paradox, akka-actor, slick, jackson-module-scala, http4s-websocket, fs2, kxbmap-configs, zinc)`

**Comment by SethTisue** [comment]
141 green projects now! https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-jdk9-integrate-community-build/310/console has `failed: paradox, slick, scalikejdbc, kxbmap-configs, specs2-more, fs2, github4s, pureconfig, http4s-websocket, scala-debugger, twitter-util, zinc, akka-stream`

* akka-stream: https://github.com/scala/community-builds/pull/759 (I'm pursuing this and will consult with the Akka team as needed)
* fs2: https://github.com/functional-streams-for-scala/fs2/issues/1198 (ticket needs volunteer)

**Comment by SethTisue** [comment]
someone want to look into the specs2-more and zinc failures...? or any of the others?

**Comment by SethTisue** [comment]
* akka-stream and akka-more are green
* fs2 is green now
* Play fails. ticket: https://github.com/playframework/playframework/issues/8555
* there are Slick test failures, see https://github.com/slick/slick/issues/1908
* http4s-websocket compilation failures. ticket: https://github.com/http4s/http4s-websocket/issues/12

**Comment by SethTisue** [comment]
the expected-to-fail list is now kept in this repo itself, here: https://github.com/scala/community-builds/blob/2.12.x/report/Report.scala#L38

**Comment by SethTisue** [comment]
PR to fix blaze: https://github.com/http4s/blaze/pull/213

we are using a frozen blaze so we won't immediately get the fix once this is merged. but perhaps http4s has moved on and could handle blaze master now, I should try it and see

**Comment by SethTisue** [comment]
ticket on zinc failure: https://github.com/sbt/zinc/issues/567

**Comment by SethTisue** [comment]
ticket on boopickle failure: 
https://github.com/suzaku-io/boopickle/issues/105

**Comment by SethTisue** [comment]
> perhaps http4s has moved on and could handle blaze master now, I should try it and see

nope, we had to keep blaze frozen for now. I could re-fork it and cherry-pick 
https://github.com/http4s/blaze/pull/213
